internal/webhooks/cluster.go

```diff
@@ -69,12 +70,26 @@ func (webhook *Cluster) Default(_ context.Context, obj runtime.Object) error {
 		cluster.Spec.ControlPlaneRef.Namespace = cluster.Namespace
 	}
 
-	// If the Cluster uses a managed topology
+	// Additional defaulting if the Cluster uses a managed topology.
 	if cluster.Spec.Topology != nil {
-		// tolerate version strings without a "v" prefix: prepend it if it's not there
+		// Tolerate version strings without a "v" prefix: prepend it if it's not there.
 		if !strings.HasPrefix(cluster.Spec.Topology.Version, "v") {
 			cluster.Spec.Topology.Version = "v" + cluster.Spec.Topology.Version
 		}
+
+		// If the linked ClusterClass defines defaults for variables add them to the Cluster Topology.
+		clusterClass, errs := webhook.getClusterClassForCluster(ctx, cluster)
```

Comment:
I wonder if we should do this similar to `getClustersUsingClusterClass`, i.e. return a simple error and then build the field error outside of the call, to have those two functions consistent regarding error handling

I think an additional argument in favor is that those two funcs will probably end up in an exported util over time and in that case returning a regular error is better.

Reply:
One additional reason. The error message we currently have in `getClusterClassForCluster` doesn't fit for both use cases, but this would be automatically solved if we build the error outside of `getClusterClassForCluster`.

Reply:
That makes sense I think.